### ISSUE TITLE
t/dbic.t fails (with -Duselongdouble?)

### ISSUE THREAD

**Issue by eserte** [GrantStreetGroup/DBIx-BatchChunker#3]
On some of my smoker systems the test suite fails like this:
```
# Failed test 'Subtest: DBIC Processing (+ process_past_max)'
# at t/dbic.t line 88.
# Caught exception in subtest: Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"}) at /home/cpansand/.cpan/build/2021012820/DBIx-BatchChunker-v0.940.2-1/blib/lib/DBIx/BatchChunker.pm line 1016
#     Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"})
#     "Int" is defined as: (do { my $tmp = $_; defined($tmp) and !ref($tmp) and $tmp =~ /\A-?[0-9]+\z/ })

# Failed test 'Subtest: DIY Processing (+ process_past_max)'
# at t/dbic.t line 163.
# Caught exception in subtest: Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"}) at /home/cpansand/.cpan/build/2021012820/DBIx-BatchChunker-v0.940.2-1/blib/lib/DBIx/BatchChunker.pm line 1016
#     Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"})
#     "Int" is defined as: (do { my $tmp = $_; defined($tmp) and !ref($tmp) and $tmp =~ /\A-?[0-9]+\z/ })

# Failed test 'Subtest: process_past_max + min_chunk_percent'
# at t/dbic.t line 217.
# Caught exception in subtest: Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"}) at /home/cpansand/.cpan/build/2021012820/DBIx-BatchChunker-v0.940.2-1/blib/lib/DBIx/BatchChunker.pm line 1016
#     Value "1.84467440737095516e+19" did not pass type constraint "Int" (in $self->{"max_end"})
#     "Int" is defined as: (do { my $tmp = $_; defined($tmp) and !ref($tmp) and $tmp =~ /\A-?[0-9]+\z/ })
t/dbic.t ................. 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/9 subtests 
```

Looking at http://matrix.cpantesters.org/?dist=DBIx-BatchChunker%20v0.940.2;reports=1 there are more fail reports, and it seems that most of them have in common to run on a perl with increased floating point accuracy (either *-ld (longdouble) or *-quadmath perls).

**Comment by SineSwiper** [comment]
I'm rather puzzled how `Data::Float::max_integer`, which is supposed to represent the maximum integer that can be safely counted, ends up turning into an unsafe and unwanted scientific notation string.

Okay, I'm able to recreate it with the help of a newly-built plenv Perl.  Just need to force it to be a UV.

**Comment by SineSwiper** [comment]
Fixed in v0.940.3.
